### PR TITLE
fix: (Core) Overflowing and stacked tabs

### PR DESCRIPTION
**Pull request by salarenko**
#### Please provide a link to the associated issue.
Closes: #1410 

#### Please provide a brief summary of this pull request.
This PR:
- Adds collapsible tabs functionality
- Adds maximum number of visible tabs functionality
- Adds adds stacked tabs content functionality
- Adds `getElementCapacity` and `getElementWidth` utils for counting elements capacity and width (with margins)
- Adds a `[scrollSpyDisabled]` attribute to disable `ScrollSpy`

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/tabs/examples/tabs-example.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <label fd-form-label>Simple Tabs Built by Passing Title Parameter</label>
-<fd-tab-list>
+<fd-tab-list [maxVisibleTabs]="2">
```
Comment (JKMarkowski) [inline apps/docs/src/app/core/component-docs/tabs/examples/tabs-example.component.html:2]: Is it necessary?

Comment (salarenko) [reply apps/docs/src/app/core/component-docs/tabs/examples/tabs-example.component.html]: Nope, removed.

File: libs/core/src/lib/tabs/tab-list.component.ts [hunk 2]
```diff
@@ -179,17 +246,147 @@ export class TabListComponent implements AfterViewInit, OnChanges, OnDestroy {
...
+            this._panelTabsArray = panelTabs;
+            this._tabs = { visible: panelTabs, overflowing: [] };
+        })
+    }
+
+    private _listenOnPanelTabsChangeAndCollapse(): void {
+        const $tabItemsSource = this.tabItems.changes.pipe(
+            map(tabItems => tabItems.toArray()),
+            first()
+        );
```
Comment (JKMarkowski) [inline libs/core/src/lib/tabs/tab-list.component.ts:381]: Is there any reason to have this observable?

**Review by JKMarkowski — COMMENTED** [review]
Only one thing, rest LGTM. Great Job!

File: libs/core/src/lib/tabs/tab-list.component.ts (hunk 2 above)
Comment (salarenko) [reply libs/core/src/lib/tabs/tab-list.component.ts]: Yes, there is a following flow in the component:
`TabPanel(ContentProjected)` -> `TabItem`

So tab headers are build based on the tab panels. This means that when the tab panels change, tab headers are still not updated. That's why on tab panels change I subscribe to tab headers change and run collapse logic.

**Review by JKMarkowski — APPROVED** [review]

**Review by katekozlowska — APPROVED** [review]
LGTM, good job! :)

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
- [ ] When you select a tab from the overflow menu, this tab is taken out of the overflow menu and put as last from the "visible" tabs.
Screen Shot 2020-12-09 at 4 30 14 PM

Expected behaviour:
Screen Shot 2020-12-09 at 4 31 45 PM

- [ ] The button should be customizable (could add text to the arrow, like "More" or "Show More", etc.)
Screen Shot 2020-12-09 at 4 33 18 PM

UX guidelines:
Screen Shot 2020-12-09 at 4 36 15 PM

- [ ] there's no indication that a tab is selected (missing the underline)
Screen Shot 2020-12-09 at 4 39 22 PM

- [ ] why is the active state in grey?
Screen Shot 2020-12-09 at 4 40 13 PM

How it should be:
Screen Shot 2020-12-09 at 4 41 00 PM

**Comment by droshev** [comment]
Regarding the missing selected tab indication - most of the examples are having this issue

**Comment by droshev** [comment]
a question from my side: when we configure to limit the number of visible tabs shall we put the overflow menu next to the last visible tab?

**Comment by salarenko** [comment]
@InnaAtanasova 

> When you select a tab from the overflow menu, this tab is taken out of the overflow menu and put as last from the "visible" tabs.
- This behavior is not described in Fiori Design Guidelines, but I think it makes sense, I will add it.

> There's no indication that a tab is selected (missing the underline)
- This is some other issue not associated with this PR

>  why is the active state in grey?
- This is just an example. Because default "selected indicator" is broken, I added neutral state to these tabs to show the indicator.

@droshev 
> when we configure to limit the number of visible tabs shall we put the overflow menu next to the last visible tab?
- This is probably a question to the designers. Limited number of visible tabs is a custom thing not described in Fiori Design Guidelines. Currently the overflow menu is placed as described in the guidelines.

**Comment by droshev** [comment]
> * er issue not associated with this PR

@salarenko 
Let's fix it even if the issue is not related with the PR

**Pull request closed without merging**